SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -99,6 +99,11 @@ void USpatialReceiver::OnAddComponent(Worker_AddComponentOp& Op)
 	UE_LOG(LogSpatialReceiver, Verbose, TEXT("AddComponent component ID: %u entity ID: %lld"),
 		Op.data.component_id, Op.entity_id);
 
+	if (bInCriticalSection)
+	{
+		StaticComponentView->OnAddComponent(Op);
```

Comment:
Do we know this is true? It is definitely not possible to receive an authority delegation for another entity during a critical section?

Reply:
I think this will be better if we check if the component is in the view. Which is also what the worker wrapper does.

Reply:
We should talk about this bit in person.